extension/content/engineUrlView.mjs

```diff
@@ -43,21 +50,30 @@ export default class EngineUrlView extends HTMLElement {
     );
 
     // Create body rows
-    rowHeaders.forEach((header, index) => {
+    rowHeaders.forEach((header) => {
       let row = tbody.insertRow(-1);
       this.createAndAppendCell(row, "th", header);
 
       let cell = row.insertCell(-1);
-      let url = sortedUrls[index];
+      cell.textContent = "";
+    });
+
+    fragment.appendChild(this.#table);
+    this.appendChild(fragment);
+  }
+
+  updateUrls(header, sortedUrls) {
```

Comment:
For `updateUrls` function, could we split it up into two? Reason: updating the header does not update a url, it only updates the engine name. 

Maybe something like this to make it more clear: 
```js
#updateHeader(header) {
  this.#table.tHead.rows[0].cells[1].textContent = header;
}

#updateUrls(sortedUrls) {
  let tBody = this.#table.tBodies[0];

  sortedUrls.forEach((url, index) => {
    let cell = tBody.rows[index].cells[1];
    if (url) {
      cell.replaceChildren(this.createAnchor(url));
    } else {
      cell.textContent = "Not Specified";
    }
  });
}
```

Reply:
I'm not sure there's a need to split this into two, I do think I can improve the function naming though.